### PR TITLE
Improve configgen command line handling a bit

### PR DESCRIPTION
**Pull request by dpc**
Positional arguments are not descriptive and harder to use.

### CODE REVIEW

**Comment by dpc** [comment]
Feedback (especially w.r.t naming) much appreciated.

**Comment by DanGould** [comment]
Concept ACK. Much better! Consuming this script right now as well.

It makes sense for these values to have sane defaults as well. Perhaps the ones in build.sh make sense. I'm not sure what "hbbft" and "api" mean, but I'm just getting up to speed so maybe those names are good.

**Comment by DanGould** [comment]
Also, what unit are the denominations? msat? That should be labeled too

**Comment by dpc** [comment]
Ports and denominations should have a default.

**Comment by DanGould** [comment]
fyi esplora uses default ports 5000 & 5001. Best to avoid those. Esplora pairs nicely with fedimint.

**Comment by dpc** [comment]
There's 63k ports to choose from and it always have to be 8000, 8080, 3000 or some other x000. :shrug:

File: fedimint/src/bin/configgen.rs [hunk 1]
```diff
@@ -5,22 +5,41 @@ use fedimint_api::{Amount, PeerId};
...
+    #[clap(long = "hbbft-base-port")]
     hbbft_base_port: u16,
+
+    /// Base api port
+    #[clap(long = "api-base-port")]
     api_base_port: u16,
-    amount_tiers: Vec<Amount>,
+
+    /// Available denominations of notes issues by the federation (comma separated)
+    #[clap(long = "denominations", value_delimiter = ',')]
```
Comment (elsirion) [inline fedimint/src/bin/configgen.rs:32]: Can we enforce that there has to be at least one denomination? Forgetting to specify them bit me a few times …

I would have assumed that this is the default behavior, but apparently it isn't (I just tested it again) and there has been a lengthy discussion about it (didn't read it all) https://github.com/clap-rs/clap/issues/1772

**Review by elsirion — CHANGES_REQUESTED** [review]
Really cool! Naming looks good to me. Left one idea I'd like to discuss.

**Comment by dpc** [comment]
`--denominations` are now required with at least on element, and ports get some randomish default values.

File: fedimint/src/bin/configgen.rs [hunk 2]
```diff
@@ -5,22 +5,46 @@ use fedimint_api::{Amount, PeerId};
...
+    /// Directory to output all the generated config files
+    #[clap(long = "out-dir")]
+    dir_out_path: PathBuf,
+
+    /// Number of nodes in the federation
+    #[clap(long = "num-nodes")]
+    num_nodes: u16,
+
+    /// Base hbbft port
+    #[clap(long = "hbbft-base-port", default_value = "17240")]
```
Comment (elsirion) [inline fedimint/src/bin/configgen.rs:24]: I checked that the port is out of the range for ephemeral ports on most systems (32768–60999 for Linux apparently).

**Review by elsirion — APPROVED** [review]

**Pull request merged**